src/plugins/flexibleLayout/plugin.js

```diff
@@ -0,0 +1,138 @@
+/*****************************************************************************
+ * Open MCT, Copyright (c) 2014-2018, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define([
+    './flexibleLayout'
+], function (
+    FlexibleLayout
+) {
+    return function plugin() {
+
+        return function install(openmct) {
+            openmct.objectViews.addProvider(new FlexibleLayout(openmct));
+
+            openmct.types.addType('flexible-layout', {
+                name: "Flexible Layout",
+                creatable: true,
+                description: "A fluid, flexible layout canvas that can display multiple objects in rows or columns.",
+                cssClass: 'icon-flexible-layout',
+                initialize: function (domainObject) {
+                    domainObject.configuration = {
+                        containers: [],
+                        rowsLayout: false
+                    };
+                }
+            });
+
+            openmct.toolbars.addProvider({
+                name: "Flexible Layout Toolbar",
+                key: "flex-layout",
+                description: "A toolbar for objects inside a Flexible Layout.",
+                forSelection: function (selection) {
+                    let parent = selection[selection.length - 1];
+
+                    return (parent && parent.context.item &&
+                        parent.context.item.type === 'flexible-layout' &&
+                        openmct.editor.isEditing());
+                },
+                toolbar: function (selection) {
+                    let domainObject = selection[selection.length - 1].context.item;
+
+                    let deleteButton = {
```

Comment:
creating each control based on selection[0] (which could be a frame, container, or flex layout) required you to pass the full context to each child component.  This complicates child components with information they don't need to know,  You can use the selection path to simplify code considerably:

let toolbar = [];
if selection[0] is a frame, push deleteFrame button.
if selection[0] is a container, push deleteContainer button
if selection[0] is a flex layout, push addContainer button.  
if selection[1] or selection[2] is a flex layout, push addContainer button and read containerIndex from selection[0] or selection[1].